renderdoc/driver/gl/gl_driver.cpp

```diff
@@ -739,6 +742,50 @@ std::string WrappedOpenGL::GetChunkName(uint32_t idx)
   return ToStr((GLChunk)idx);
 }
 
+ResourceId WrappedOpenGL::GetContextResourceID()
+{
+  uint64_t currentId = Threading::GetCurrentID();
+  if(m_ActiveContextResourceIDs.find(currentId) == m_ActiveContextResourceIDs.end())
+  {
+    m_ActiveContextResourceIDs[currentId] =
+        GetResourceManager()->RegisterResource(GLResource(NULL, eResSpecial, eSpecialResContext));
```

Comment:
This will give all the contexts an identical `GLResource` but register it multiple times - this might work in some cases but probably will have some edge cases where it breaks. It's an easy fix - when making the other refactors, pass the `ctx` pointer through instead of `NULL` as the first parameter, to make each `GLResource` context-specific.